src/SignInWithKlarna.php

```diff
@@ -0,0 +1,119 @@
+<?php
+namespace Krokedil\SignInWithKlarna;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Sign_In_With_Klarna class.
+ */
+class SignInWithKlarna {
+
+	/**
+	 * The handle name for the JavaScript library.
+	 *
+	 * @var string
+	 */
+	public static $library_handle = 'siwk_library';
+
+	/**
+	 * The action hook name for outputting the placement HTML.
+	 *
+	 * @var string
+	 */
+	public static $siwk_placement_hook = 'siwk_placement';
+
+	/**
+	 * The internal settings state.
+	 *
+	 * @var Settings
+	 */
+	public $settings;
+
+	/**
+	 * Class constructor.
+	 *
+	 * @param array $settings The plugin settings to extract from.
+	 */
+	public function __construct( $settings ) {
+		$this->settings = new Settings( $settings );
+
+		add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_scripts' ) );
+		Ajax::init();
+	}
+
+	/**
+	 * Enqueue scripts.
+	 *
+	 * Determines whether the SIWK button should be rendered.
+	 *
+	 * @return void
+	 */
+	public function enqueue_scripts() {
+		/**
+		 * Check if we need to display the SIWK button:
+		 * 1. if logged in or guest but has not signed in with klarna.
+		 * 2. signed in, but need to renew the refresh token.
+		 */
+		$show_button = empty( get_user_meta( get_current_user_id(), User::$refresh_token_key, true ) );
+		if ( ! $show_button ) {
+			return;
+		}
+
+		$path        = dirname( __FILE__ ) . '/assets/siwk.js';
+		$script_path = substr( $path, strpos( $path, '/wp-content' ) );
```

Comment:
Id like to discuss this if possible. I just want to make sure this works as intended and wont cause issues.

Getting the URL for the assets is probably one of the most complicated things when it comes to doing this in the package, so i just want to go over possible issues if we can.